packages/core/core/generate-text/generate-text.ts

```diff
@@ -107,77 +108,87 @@ By default, it's set to 0, which will disable the feature.
      */
     maxToolRoundtrips?: number;
   }): Promise<GenerateTextResult<TOOLS>> {
-  const retry = retryWithExponentialBackoff({ maxRetries });
-  const validatedPrompt = getValidatedPrompt({ system, prompt, messages });
-
-  const mode = {
-    type: 'regular' as const,
-    ...prepareToolsAndToolChoice({ tools, toolChoice }),
-  };
-  const callSettings = prepareCallSettings(settings);
-  const promptMessages = convertToLanguageModelPrompt(validatedPrompt);
-
-  let currentModelResponse: Awaited<ReturnType<LanguageModel['doGenerate']>>;
-  let currentToolCalls: ToToolCallArray<TOOLS> = [];
-  let currentToolResults: ToToolResultArray<TOOLS> = [];
-  let roundtrips = 0;
-  const responseMessages: Array<CoreAssistantMessage | CoreToolMessage> = [];
-
-  do {
-    currentModelResponse = await retry(() => {
-      return model.doGenerate({
-        mode,
-        ...callSettings,
-        // once we have a roundtrip, we need to switch to messages format:
-        inputFormat: roundtrips === 0 ? validatedPrompt.type : 'messages',
-        prompt: promptMessages,
-        abortSignal,
+  const tracer = await getTracer();
+  return tracer.startActiveSpan('generateText', async span => {
```

Comment:
Name should probably be a bit more indicative. E.g. `ai.generateText`? Also, good idea to add a bunch of attributes, including `model`, `size-of-prompt`, etc.